packages/cozy-client/src/store/index.js

```diff
@@ -72,9 +72,12 @@ const combinedReducer = (state = { documents: {}, queries: {} }, action) => {
       )
     }
   }
+  const updatedDocuments = documents(state.documents, action)
+  const documentsChanged = updatedDocuments !== state.documents
+
   return {
-    documents: documents(state.documents, action),
-    queries: queries(state.queries, action, state.documents)
+    documents: updatedDocuments,
+    queries: queries(state.queries, action, state.documents, documentsChanged)
```

Comment:
> Is it more wise to pass, instead of documentsChanged, the previous state?

We could and it would work, but I'm not sure it's better..? The responsibility of the `queries` reducer is to update whatever queries have changed as a result of that action. To do this, we need to give it some information — the action and the current state of course, but also the state of `documents` for legacy reason, and now whether or not these documents have *actually* changed. I don't think it should be the responsibility of `queries` to check if they have changed or not. Does that make sense ? or did I misunderstand something ?

(If we decide to keep the boolean I agree we should rename it to something more obvious 👍 )

Reply:
If the document have not changed, we can completely bypass `queries` right ?

```js
    queries: !documentsChanged ? state.queries : queries(state.queries, action, state.documents)
```

Reply:
Sadly not, because for actions such as `INIT_QUERY` we still need to go through the reducer. Even when the actions are of the right type, some queries might need an update while some others may be fine, depending on their status.

Reply:
OK thanks.

1. I think we should pass the updatedDocuments to `queries` otherwise the `manualUpdater` from `queries` could potentially receive out-of-date documents.

(It could be named `nextDocuments` to echo the nextState/nextProps in React).

2. We have 2 solutions afterwards to check if documents have changed : pass the complete state and let queries do `state.documents === nextDocuments` or have the boolean variable `haveDocumentsChanged`. Passing the complete `state` to `queries` seems a bit "dangerous" to me as then `queries` needs to know about the structure of the state. I am in favor of the boolean even if I am not so happy about it.

Reply:
For this use case (ie nothing changed), a boolean is certainly good enough. 

But how we'll do this kind of check, if only one document has changed ? Does the `updater()` will need the previous state?

Reply:
> But how we'll do this kind of check, if only one document has changed ? Does the updater() will need the previous state?

I see what you mean, but I don't think we'll need this kind of checks, ever (of cour se when you say stuff like this, you are eventually proven wrong…). I don't see a use case here the query reducer needs to know how many documents have changed and behave differently according to that. I mean maybe I'll be wrong about this, but we can change the code when it happens? It's a purely internal modification.